src/App.tsx

```diff
@@ -14,11 +15,57 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+export class App extends React.Component {
+  state = {
+    goods: goodsFromServer,
```

Comment:
better to create a copy
```suggestion
    goods: [...goodsFromServer],
```